### PR TITLE
Git API Wrapper Tool

### PR DESCRIPTION
**Pull request by boonhapus**

### CODE REVIEW

File: cs_tools/cli/tools/git/config.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+app = CSToolsApp(
+    name="config",
+    help="Tools for working with git configurations.",
+    add_completion=False,
+    invoke_without_command=True,
+    no_args_is_help=True,
+)
+
+
+def _get_token(ctx: typer.Context) -> str:
```
Comment (boonhapus) [inline cs_tools/cli/tools/git/config.py:24]: @billdback-ts this doesn't get called, what's it supposed to do?

File: cs_tools/cli/tools/git/config.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+
+def _show_configs_as_table(configs: list[dict], title: str = "Configuration Details"):
+    """
+    Show the configurations as a table.
+    """
+    use_cnt = len(configs) > 1
+
+    cnt: int = 0
+    for _ in configs:
```
Comment (boonhapus) [inline cs_tools/cli/tools/git/config.py:185]: `enumerate` gives you this incremental counting

```python
    use_count = len(configs) > 1

    for count, config in enumerate(configs, start=1):
        title = f"{title} {count}" if use_count else title
        table = Table(title=f"Configuration Details {count}", width=100)
        table.add_column("Property", width=25)
        table.add_column("Value", width=75)

        for k, v in config.items():
            table.add_row(k, str(v))

        rich_console.print(Align.center(table))
```

File: cs_tools/cli/tools/git/branches.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        org: str = typer.Option(None, help="the org to use if any"),
+        tag: str = typer.Option(None, help="the tag for metadata to commit"),
+        metadata_ids: str = typer.Option("",
+                                     custom_type=MultipleChoiceType(),
+                                     help="the metadata GUIDs or names to commit"),
+        branch_name: str =
+        typer.Option(None,
+                     help="the branch name to use for the git repository (or use the default if not provided"),
+        comment: str = typer.Option(..., help="the comment to use for the commit"),
+        delete_aware: bool = typer.Option(False, help="deletes content that doesn't exist in TS from the repo"),
```
Comment (boonhapus) [inline cs_tools/cli/tools/git/branches.py:35]: This argument doesn't get used, what's it supposed to do?

File: cs_tools/cli/tools/git/config.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+    except HTTPStatusError as e:
+        rich_console.print(f"[bold red]Error creating the configuration: {e.response}.[/]")
+        rich_console.print(f"[bold red]{e.response.content}.[/]")
+
+
+@app.command(dependencies=[thoughtspot], name="update")
+def config_update(
+        ctx: typer.Context,
+        repository: str = typer.Option(None, help="the git repository to use"),
```
Comment (boonhapus) [inline cs_tools/cli/tools/git/config.py:81]: This argument doesn't get used, what's it supposed to do?

File: cs_tools/cli/tools/git/config.py (hunk 1 above)
Comment (billdback-ts) [reply cs_tools/cli/tools/git/config.py:24]: Ooops.  That's one I left in when I thought we needed the token.  Removing.

File: cs_tools/cli/tools/git/config.py (hunk 2 above)
Comment (billdback-ts) [reply cs_tools/cli/tools/git/config.py:185]: Oh, that's good to know.  Svelte does this as well.  Changing.

File: cs_tools/cli/tools/git/branches.py (hunk 3 above)
Comment (billdback-ts) [reply cs_tools/cli/tools/git/branches.py:35]: It checks allows commits to remove things in GitHub.  It was in the v2 API, but wasn't getting passed.  Added.

File: cs_tools/cli/tools/git/config.py (hunk 4 above)
Comment (billdback-ts) [reply cs_tools/cli/tools/git/config.py:81]: Removed.  Not a valid argument to update (for some reason - guess the expectation is it won't change).

**Review by billdback-ts — COMMENTED** [review]
OK, looks good for a first version.

**Pull request merged**